lib/fizzy/execute.cpp

```diff
@@ -566,7 +574,7 @@ execution_result execute(Instance& instance, FuncIdx func_idx, std::vector<uint6
     assert(code_idx < instance.module.codesec.size());
 
     const auto& code = instance.module.codesec[code_idx];
-    auto& memory = *instance.memory;
+    auto* const memory = instance.memory.get();
```

Comment:
Why this?

Reply:
Why pointer in general? Because you cannot dereference null pointer.
Why `auto* const`? The `*` indicates it is a pointer. The `const` make it const. More verbose variant is `const auto`.

Reply:
> Why pointer in general? Because you cannot dereference null pointer.

I don't get what this fixes, you just dereference it many times instead of once now.

Reply:
Let's move this discussion to: https://github.com/wasmx/fizzy/pull/225.

Reply:
Ah I see now, in general module is allowed to have no memory, unless it has some memory instructions.

Reply:
So this PR as-is can be applied after #225 is merged?